Dask on Ray incompatible with dask==2021.3.1

Attempting to import `ray.util.dask.ray_dask_get` with `dask==2021.3.1` raises the following error:

```
Traceback:
../../.pyenv/versions/3.7.8/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/integration_tests/test_ray.py:23: in <module>
    from ludwig.backend.ray import RayBackend, get_horovod_kwargs
ludwig/backend/ray.py:24: in <module>
    from ray.util.dask import ray_dask_get
env/lib/python3.7/site-packages/ray/util/dask/__init__.py:2: in <module>
    from .scheduler import ray_dask_get, ray_dask_get_sync
env/lib/python3.7/site-packages/ray/util/dask/scheduler.py:15: in <module>
    from .common import unpack_object_refs
env/lib/python3.7/site-packages/ray/util/dask/common.py:10: in <module>
    from dask.compatibility import apply, is_dataclass, dataclass_fields
E   ImportError: cannot import name 'is_dataclass' from 'dask.compatibility' (/Users/tgaddair/repos/ludwig/env/lib/python3.7/site-packages/dask/compatibility.py)
```

This is because Dask dropped these utility methods as part of https://github.com/dask/dask/pull/7006.

Dask on Ray should be updated for the new API.

--- a/python/ray/util/dask/common.py
+++ b/python/ray/util/dask/common.py
@@ -7,7 +7,18 @@ import ray
 
 from dask.base import quote
 from dask.core import get as get_sync
-from dask.compatibility import apply, is_dataclass, dataclass_fields
+from dask.compatibility import apply
+
+
+try:
+    from dataclasses import is_dataclass, fields as dataclass_fields
+except ImportError:
+    # Python < 3.7
+    def is_dataclass(x):
+        return False
+
+    def dataclass_fields(x):
+        return []
 
 
 def unpack_object_refs(*args):

--- a/python/ray/util/dask/common.py
+++ b/python/ray/util/dask/common.py
@@ -9,7 +9,6 @@ from dask.base import quote
 from dask.core import get as get_sync
 from dask.compatibility import apply
 
-
 try:
     from dataclasses import is_dataclass, fields as dataclass_fields
 except ImportError:

